module/behaviour/strategy/PS3Walk/src/PS3Walk.cpp

```diff
@@ -17,123 +17,144 @@
-                                    else {
-                                        NUClear::log("Head locked");
-                                    }
-                                    headLocked = !headLocked;
-                                }
-                                break;
-                            /*case BUTTON_L1:
-                                if (event.value > 0) { // button down
-                                    NUClear::log("Requesting Left Side Kick");
-                                    emit(std::make_unique<KickScriptCommand>(KickScriptCommand{
-                                        {0, -1, 0}, // vector pointing right relative to robot
-                                        LimbID::LEFT_LEG
-                                    }));
-                                }
-                                break;
-                            case BUTTON_L2:
-                                if (event.value > 0) { // button down
-                                    NUClear::log("Requesting Left Front Kick");
-                                    emit(std::make_unique<KickScriptCommand>(KickScriptCommand{
-                                        {1, 0, 0}, // vector pointing forward relative to robot
-                                        LimbID::LEFT_LEG
-                                    }));
-                                }
-                                break;
-                            case BUTTON_R1:
-                                if (event.value > 0) { // button down
-                                    NUClear::log("Requesting Right Side Kick");
-                                    emit(std::make_unique<KickScriptCommand>(KickScriptCommand{
-                                        {0, 1, 0}, // vector pointing left relative to robot
-                                        LimbID::RIGHT_LEG
-                                    }));
-                                }
-                                break;*/
-                            case BUTTON_R2:
-                                if (event.value > 0) {  // button down
-                                    NUClear::log("Requesting Right Front Kick");
-                                    emit(std::make_unique<KickScriptCommand>(KickScriptCommand(
-                                        Eigen::Vector3d(1, 0, 0),  // vector pointing forward relative to robot
-                                        LimbID::RIGHT_LEG)));
-                                }
-                                break;
-                        }
+                    else {
+                        log<NUClear::INFO>("Start walking");
+                        moving = true;
                     }
                 }
-                // If it's closed then we should try to reconnect
-                else if (!joystick.valid()) {
-                    joystick.reconnect();
+            });
+
+            on<Trigger<HeadControlEvent>>().then([this](const HeadControlEvent& event) {
```

Comment:
I'm not sure exactly how I feel about naming these messages things like `HeadControlEvent`. On one hand it means if you change what event triggers this it's one place to change. On the other hand, looking at this code I have no idea what button to push to make it happen without scrolling to the top of the file.

Reply:
There's just no pleasing you, is there